api/client/client.go

```diff
@@ -0,0 +1,96 @@
+package client
+
+import (
+	"context"
+	"io"
+
+	"github.com/andres-teleport/overseer/api"
+	"github.com/andres-teleport/overseer/api/authentication"
+	"google.golang.org/grpc"
+)
+
+type Client struct {
+	client api.JobworkerServiceClient
+}
+
+func NewClient(serverAddr, keyFile, certFile, caFile string) (*Client, error) {
+	creds, err := authentication.NewClientTransportCredentials(keyFile, certFile, caFile)
+	if err != nil {
+		return nil, err
+	}
+
+	conn, err := grpc.Dial(serverAddr, grpc.WithTransportCredentials(creds))
+	if err != nil {
+		return nil, err
+	}
+
+	return &Client{client: api.NewJobworkerServiceClient(conn)}, nil
+}
+
+func (c *Client) Start(command string, arguments ...string) (string, error) {
+	job := &api.Job{
+		Command:   command,
+		Arguments: arguments,
+	}
+
+	jobID, err := c.client.Start(context.Background(), job)
+	if err != nil {
+		return "", err
+	}
+
+	return jobID.Id, nil
+}
+
+func (c *Client) Stop(jobID string) error {
+	_, err := c.client.Stop(context.Background(), &api.JobID{Id: jobID})
+	return err
+}
+
+func (c *Client) Status(jobID string) (*api.StatusResponse, error) {
+	return c.client.Status(context.Background(), &api.JobID{Id: jobID})
+}
+
+func copyStream(streamer api.JobworkerService_StdOutClient, w *io.PipeWriter) {
+	for eof := false; !eof; {
```

Comment:
Nit: Might be a bit cleaner to modify the condition to `for err != io.EOF`, wouldn't need additional variable.

Reply:
I tried but I'm not convinced, it's roughly the same number of lines but I have to be extra careful to avoid overwriting the error by mistake, so it seems more error-prone.

It looks like this:
```go
	var err error
	var chunk *api.OutputChunk
	for err != io.EOF {
		chunk, err = streamer.Recv()
		if err != nil && err != io.EOF {
			_ = w.CloseWithError(err)
		}

		if chunk != nil {
			if _, err := w.Write(chunk.Output); err != nil {
				_ = w.CloseWithError(err)
				return
			}
		}
	}
```

Please let me know if I'm missing anything.